mrmustard/physics/representations.py

```diff
@@ -784,6 +808,15 @@ def reduce(self, shape: Union[int, Iterable[int]]) -> Fock:
         ret = self.array[(slice(0, None),) + tuple(slice(0, s) for s in shape)]
         return Fock(array=ret, batched=True)
 
+    def serialize(self) -> Path:
+        r"""Serialize a Fock instance."""
+        return save(type(self), arrays={"array": self.array})
+
+    @classmethod
+    def deserialize(cls, data) -> Fock:
+        r"""Deserialize a Fock instance."""
+        return cls(data["array"], batched=True)
```

Comment:
Not knowing how batching works, just noticing the `__init__` has `batched=False`. Would this be able to work for unbatched stuff?

Reply:
if you create it with `batched=False`, it just [adds a batch dim later in construction](https://github.com/XanaduAI/MrMustard/blob/4085e252ff8353e7769133ef7e74fedd88221766/mrmustard/physics/ansatze.py#L1052), so the internal representation is always batched. This also means that the serialized data is always batched, so we deserialize accordingly.